### PR TITLE
Server google oauth management & client temporary implementation

### PR DESCRIPTION
**Pull request by prastoin**
related to https://github.com/AdonisEnProvence/MusicRoom/issues/544

### CODE REVIEW

File: packages/client/screens/SigningInScreen.tsx [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
+    const [request, response, promptAsync] = Google.useAuthRequest({
+        expoClientId:
+            '326703248925-qg0i9ig0b4gqvkpmpfd665d5un082adj.apps.googleusercontent.com',
+        iosClientId:
+            '326703248925-n6knfko9tffasgpq3ffbv8i8qj7vpcgg.apps.googleusercontent.com',
+        androidClientId:
+            '326703248925-fq6r2lio64q1fg0ap6hj6uvm09raf9cd.apps.googleusercontent.com',
+        webClientId:
+            '326703248925-sn4nvaq3vkjoc1chuju6binov8ha3jl3.apps.googleusercontent.com',
+    });
```
Comment (Devessier) [inline packages/client/screens/SigningInScreen.tsx:35]: **👏 praise:** ‏I'm happy to see how Expo solves social authentication! I can't wait to see how this hook can be used with XState.

**Review by Devessier — DISMISSED** [review]
Great job! I am glad to see how social authentication can be handled with Expo.
You did a great job on the backend, zod is a really nice tool to check the response of an external API.
I made some minor suggestions but we can already merge.

File: packages/server/app/Models/User.ts [hunk 2]
```diff
@@ -125,14 +125,17 @@ export default class User extends BaseModel {
...
     @column()
     public rememberMeToken?: string;
 
+    @column({ columnName: 'google_id' })
+    public googleID: string | null;
+
     @beforeSave()
     public static async hashPassword(user: User): Promise<void> {
-        if (user.$dirty.password) {
+        if (user.$dirty.password && user.password) {
```
[the remark this answers is not in the document]
Comment (prastoin) [reply packages/server/app/Models/User.ts:138]: Typescript would throw an error there as given password could now be undefined

**Review by Devessier — APPROVED** [review]
We can merge!

**Pull request merged**